frog/server/accountManagement.js

```diff
@@ -43,10 +44,10 @@ export const createAccount = (
       );
     } else if (!Accounts.findUserByEmail(email)) {
       const user = Meteor.user();
-
-      if (user?.isAnonymous) {
-        // checks for duplicate email and displays error on the console.
-
+      // if the user is anonymous or a legacy user we want to simply add the email and password to their
+      // account. This allows them to keep the graphs and activities they created when they didn't have an
+      // account. Else we just create a new account
+      if (user?.isAnonymous || (user.username && !isVerifiedUser())) {
```

Comment:
We should check the user we have

```suggestion
      if (userType(user) === 'Anonymous' || userType(user) === 'Legacy') {
```